src/logfile/logfile.h

```diff
@@ -0,0 +1,208 @@
+
+    float getMinForce() { return minForce; }            ///< Return min force of the logfile
+    float getMaxForce() { return maxForce; }            ///< Return max force of the logfile
+    float getMinForceIndex() { return minForceIndex; }  ///< Return timestamp of min force
+    float getMaxForceIndex() { return maxForceIndex; }  ///< Return timestamp of max force
+
+    /**
+     * @brief Write the current metadata and force vector into a file
+     *
+     * @return true if successfully written to the harddisk
+     */
+    bool write();
+
+    /**
+     * @brief Set the path to the file
+     *
+     * @param path Path to the file
+     */
+    void setPath(QString path);
+
+    /**
+     * @brief Get the path of the logfile
+     *
+     * @return QString Absolute path
+     */
+    QString getPath();
+
+    /**
+     * @brief Get the name of the logfile without the path
+     *
+     * @return QString Name of the logfile
+     */
+    QString getFileName();
+
+    /**
+     * @brief Set a new metadata
+     *
+     * @param newData Reference to the new data
+     */
+    void setMetadata(Metadata& newData);
+
+    /**
+     * @brief Get a reference to the metadata
+     *
+     * @return Metadata&
+     */
+    Metadata& getMetadata();
+
+    /**
+     * @brief Get the forceVector
+     *
+     * @return QVector<float>&
+     */
+    QVector<float>& getForce();
+
+    /**
+     * @brief Get the timeVector
+     *
+     * @return QVector<float>&
+     */
+    QVector<float>& getTime();
+
+    /**
+     * @brief Set the forceVector
+     *
+     * @param force
+     */
+    void setForce(QVector<float>& force);
+
+    /**
+     * @brief Set the timeVector
+     *
+     * @param time
+     */
+    void setTime(QVector<float>& time);
+
+   private:
+    /**
+     * @brief Parse the metadata from a given text stream
+     *
+     * @param in QTextStream of the logfile
+     * @return int 0 on success; first invalid line number on failure
+     */
+    int parseMetadata(QTextStream& in);
+
+    /**
+     * @brief Split the input line and return a float
+     *
+     * @tparam T Either int or float
+     * @param input QString to parse to float
+     * @param fieldName QString with the field name (in front of the = sign)
+     * @param success Ptr to success bool
+     * @return T Parsed value
+     */
+    template <typename T>
+    T parseNumericField(const QString& input, const QString& fieldName, bool* success) {
+        QStringList splitList = input.split("=");
+        *success = false;
+        if (splitList.count() != 2) {
+            return -1;
+        }
+
+        if (splitList[0] != fieldName) {
+            return -1;
+        }
+
+        // Type depending code for int and float
+        if constexpr (std::is_same_v<T, int>) {
+            return splitList[1].toInt(success);
+        } else if constexpr (std::is_same_v<T, float>) {
+            return splitList[1].toFloat(success);
+        } else {
+            return -1;
```

Comment:
```suggestion
           static_assert(!sizeof(T*), "T must be int or float");
```
Why this works and `static_assert(false)` doesn't, see [this](https://stackoverflow.com/questions/38304847/constexpr-if-and-static-assert).